File: FTP Client for WinForms using Ultimate FTP library/[C#]-FTP Client for WinForms using Ultimate FTP library/C#/FileMask.cs
using System;
using System.Windows.Forms;

namespace ClientSample
{
    public partial class FileMask : Form
    {
        public FileMask()
        {
            InitializeComponent();
        }

        public FileMask(string mask, bool expand)
        {
            InitializeComponent();

            txtFileType.Text = mask;

            this.Text = expand ? "Expand selection" : "Shrink selection";
        }

        public FileMask(string mask, string title)
        {
            InitializeComponent();

            txtFileType.Text = mask;

            this.Text = title;
        }

        public string Mask
        {
            get { return txtFileType.Text; }
        }

        /// <summary>
        /// Handles the OK button's Click event.
        /// </summary>
        /// <param name="sender">The button object.</param>
        /// <param name="e">The event arguments.</param>
        private void btnOk_Click(object sender, EventArgs e)
        {
            // Check for invalid characters.
            if (Mask.Contains("/") || Mask.Contains("\\"))
            {
                MessageBox.Show("Character '/' or '\\' should not be entered", "New Name Prompt", MessageBoxButtons.OK, MessageBoxIcon.Stop);
                return;
            }

            this.DialogResult = DialogResult.OK;
        }
    }
}

File: FTP Client for WinForms using Ultimate FTP library/[VB]-FTP Client for WinForms using Ultimate FTP library/VB/FileMask.vb
Namespace ClientSample
	Partial Public Class FileMask
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub New(ByVal mask As String, ByVal expand As Boolean)
			InitializeComponent()

			txtFileType.Text = mask

			If expand Then
				Me.Text = "Expand selection"
			Else
				Me.Text = "Shrink selection"
			End If
		End Sub

		Public Sub New(ByVal mask As String, ByVal title As String)
			InitializeComponent()

			txtFileType.Text = mask

			Me.Text = title
		End Sub

		Public ReadOnly Property Mask() As String
			Get
				Return txtFileType.Text
			End Get
		End Property

		''' <summary>
		''' Handles the OK button's Click event.
		''' </summary>
		''' <param name="sender">The button object.</param>
		''' <param name="e">The event arguments.</param>
		Private Sub btnOk_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnOk.Click
			' Check for invalid characters.
			If Mask.Contains("/") OrElse Mask.Contains("\") Then
				MessageBox.Show("Character '/' or '\' should not be entered", "New Name Prompt", MessageBoxButtons.OK, MessageBoxIcon.Stop)
				Return
			End If

			Me.DialogResult = System.Windows.Forms.DialogResult.OK
		End Sub
	End Class
End Namespace